pkgs/development/python-modules/diagrams/default.nix

```
@@ -0,0 +1,66 @@
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchFromGitHub
+#, poetry
+, bash
+, black
+, jinja2
+, round
+, graphviz
+, inkscape
+, imagemagick
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "diagrams";
+  version = "0.17.0";
+  format = "pyproject";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "mingrammer";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1h5i9xb4fsmy81sznvqycxl712d9k695vznkpx2wm1ki7hvz86s8";
+  };
+
+  preConfigure = ''
+    sed -i "s@#!/bin/bash@$!/${bash}/bin/bash@" autogen.sh
+    ./autogen.sh
+  '';
+
+  # IDFK about python. but regular pip results in an empty python module
+  #installPhase = ''
+    #${poetry}/bin/poetry install
+  #'';
+
+  # the deps specified here below dont exactly match what is in pyproject.toml
+  # I did not investigate whether/why the others are listed yet:
+  #
+  #[tool.poetry.dev-dependencies]
+  #pylint = "^2.4"
+  #rope = "^0.14.0"
+  #isort = "^4.3"
+
+  checkInputs = [ pytest ];
```

That seems wrong, when the checkPhase is actually using `python -m unittest`.

---

pytest is compatible, but should be
```suggestion
  checkInputs = [ pytestCheckHook ];
```
to ensure tests are running.

However, the current checkphase doesn't use it all... so ?

---

Thanks for the suggestions. Tests are running now (about to push a rebased commit).

`pytest` runs the `unittest`-based tests, after adding `pytestCheckHook` and removing the `checkPhase` (which I had duplicated from the diagrams instructions).